api/graphql/directives/protected.js

```diff
@@ -14,6 +14,16 @@ export default class ProtetedDirective extends SchemaDirectiveVisitor {
         field._parentEntity ||
         camelcase(info.parentType.name);
 
+      const postExecution = _this.args.post_execution;
+
+      if (postExecution === true) {
+        const entity = resolve.call(this, parent, _args, ctx, info);
+
+        return await AuthPolicy.can(ctx.viewer)
+          .perform(`${entity}:${field.name}:read`)
+          .on(parent);
```

Comment:
```javascript
   return await AuthPolicy.can(ctx.viewer)
          .perform(`${entityName}:${field.name}:read`)
          .having(result)
          .on(parent) ? result : null;
```

Read the docs [here](https://github.com/hereisnaman/auth-policy) to learn more.

Also create a var above so that it can be used at both places:
```javascript
const action = `${entityName}:${field.name}:read`
```